python/src/iceberg/types.py

```diff
@@ -15,61 +15,84 @@
 # specific language governing permissions and limitations
 # under the License.
 """Data types used in describing Iceberg schemas
-
 This module implements the data types described in the Iceberg specification for Iceberg schemas. To
 describe an Iceberg table schema, these classes can be used in the construction of a StructType instance.
-
 Example:
     >>> StructType(
         [
             NestedField(True, 1, "required_field", StringType()),
             NestedField(False, 2, "optional_field", IntegerType()),
         ]
     )
-
 Notes:
   - https://iceberg.apache.org/#spec/#primitive-types
 """
 
-from typing import Optional
+from typing import Any, Dict, List, Optional, Tuple
 
 
-class Type:
-    def __init__(self, type_string: str, repr_string: str, is_primitive=False):
-        self._type_string = type_string
-        self._repr_string = repr_string
-        self._is_primitive = is_primitive
+class IcebergType:
```

Comment:
What's the motivation for renaming `Type` to `IcebergType`? Does `Type` conflict with something?

Reply:
It seems we don't have this conflict based on the python_legacy and the current development. Also, this class is internal and won't be used extensively and not sure if the renaming is required.

Reply:
Definitely not required, but I felt `IcebergType` was clearer and as you say 

> this class is internal and won't be used extensively



so to me that would mean making the change for the sake of clarity and avoiding the unnecessary collision. Would this base type be touched at all even or is it just for consolidating logic?

Reply:
I switched back to just `Type`

Reply:
I'm good either way. Up to you, but now that I understand the motivation I'd probably use `IcebergType` as well.

Reply:
haha okay I'll use `IcebergType` again